docs: expand docs for SMS recovery (#2261)
ory/docs: docs/kratos/self-service/flows/account-recovery-password-reset.mdx

### Account recovery address

To start account recovery, Ory Identities must know which address to send the recovery message to. Usually this is the email
address the user provides when registering their account. Other fields inside the `traits` section are supported as well such a

> phone number to receive the code via a SMS.

Author: The identity schema can have arbitrary fields but the recovery can currently only be done via email or sms, and the mapping from identity trait to recovery channel  (email/sms) is done in the identity schema.

> If there are multiple recovery addresses registered for the user and if the feature flag `choose_recovery_address` is
> [enabled](../../../identities/get-started/account-recovery.mdx#enable-the-feature-flag-choose_recovery_address-unlocks-sending-a-recovery-code-via-sms),
> entering any of them will present them a list of (masked) addresses to choose from (up to 10). Once they have picked one address,
> and entered the address fully (to demonstrate that they indeed know the address and also to avoid information exfiltration), this
> address is used. The address can be an email address, in which case an email is sent, or a phone number, in which case a SMS is
> sent.

Reviewer: ```suggestion
```suggestion
If a user has multiple recovery addresses and the `choose_recovery_address` feature flag is [enabled](../../../identities/get-started/account-recovery.mdx#enable-the-feature-flag-choose_recovery_address-unlocks-sending-a-recovery-code-via-sms), the process is as follows:

1. The user enters any of their registered addresses to begin the recovery flow.
1. A masked list of their recovery addresses is displayed. Up to 10 addresses are shown.
1. After selecting an address, the user must enter it in full. This step verifies ownership and helps prevent information leaks.
1. The recovery message is sent to the confirmed address via email or SMS.
```
```

> If you support phone numbers in your identity schema, be sure to configure a
> [SMS HTTP gateway](../../emails-sms/10_sending-sms.mdx) so that SMSes are indeed sent.

Reviewer: ```suggestion
	An [SMS HTTP gateway](../../kratos/emails-sms/10_sending-sms.mdx) must be configured to deliver SMS messages.
```

:::info

> If the address used for recovery is the same as the address used for verification and the account isn't activated when the
> recovery flow is started, completing the recovery flow also verifies the user's address.

Reviewer: ```suggestion
Completing a recovery flow will also verify the identities address if the address is used for both recovery and verification and the identity is not yet verified when the recovery flow begins.
```

            }
          }
        },
        "telephoneNumber": {
          "type": "string",
          "format": "tel",
          "title": "Telephone Number",
          "minLength": 3,
          "ory.sh/kratos": {
            "credentials": {
              "password": {
                "identifier": true
              },
              "code": {
                "identifier": true,
                "via": "sms"
              }
            },
            "verification": {
              "via": "sms"
            },
            "recovery": {
              "via": "sms"
            }
          }
        }
      },
      "required": ["email"],
      "additionalProperties": false
    }
  }
}
```

### Attempted recovery notifications

When this option is on and users attempt to initiate recovery for unregistered addresses, the system sends an attempted recovery
notification to the email address that was used in the attempt. This prevents account enumeration attacks as explained in this
[blog post by Troy Hunt](https://www.troyhunt.com/website-enumeration-insanity-how-our-personal-data-is-leaked/). By default, this
feature is disabled in newly created Ory Network projects.

> Note that for costs reason, this concept does not exist for SMSes. That means that trying to recover an account with the wrong
> phone number will _not_ result in any sent SMS.
> 
> Follow these steps to enable sending attempted recovery notification emails:

Reviewer: ```suggestion
For cost reasons, the attempted recovery notification does not apply to SMS. An attempt to recover an account using an incorrect phone number will not trigger an SMS message.

Follow these steps to enable sending attempted recovery notification emails:
```

          "minLength": 3,
          "ory.sh/kratos": {
            "credentials": {
              "password": {
                "identifier": true
              },
              "code": {
                "identifier": true,
                "via": "sms"
              }
            },
            "verification": {
              "via": "sms"
            },
            "recovery": {
              "via": "sms"
            }
          }
        }
      },
      "required": ["email"],
      "additionalProperties": false
    }
  }
}
```

### Attempted recovery notifications

When this option is on and users attempt to initiate recovery for unregistered addresses, the system sends an attempted recovery
notification to the email address that was used in the attempt. This prevents account enumeration attacks as explained in this
[blog post by Troy Hunt](https://www.troyhunt.com/website-enumeration-insanity-how-our-personal-data-is-leaked/). By default, this
feature is disabled in newly created Ory Network projects.

For cost reasons, the attempted recovery notification does not apply to SMS. An attempt to recover an account using an incorrect
phone number will not trigger an SMS message.

> Follow these steps to enable sending attempted recovery notification emails:

Reviewer 2: The 'attempted recovery notification' noun phrase is so long... simplify the adverb to:
"Follow these steps to send attempted recovery notification emails:"

## Fallback recovery address

> In some scenarios, users may want to recover their account using a different address than the one they used to register, for
> example if they no longer have access to the original address. To use two (or more) recovery addresses, you need to define a
> secondary field, such as `email_secondary` in the traits section of the identity schema. This field will serve as the fallback
> address for account recovery. Read more about identity schemas [here](../../manage-identities/05_identity-schema.mdx).

Reviewer: ```suggestion
A user may need a fallback recovery address if they lose access to the address used for registration.

To enable this, define a secondary field, such as email_secondary, in the traits section of your [identity schema](../../manage-identities/05_identity-schema.mdx). This field will serve as the alternative address for account recovery.
```

messages to both addresses when the recovery flow is initiated. The user should provide the `email_secondary` field during
registration or update it later in their account settings so it can be used in the recovery process.

> Note that this feature is orthogonal to the situation where a user has multiple recovery addresses registered of different
> _types_, such as one email address and one phone number. In this case they will be prompted to choose which address to use during
> the recovery flow and only _one_ message will be sent to the address they picked.
> 

Reviewer: ```suggestion
This is different from the scenario where a user has multiple recovery addresses of different types, such as one email address and one phone number. In that case, the user is prompted to choose which address to use, and a recovery message is sent to only the one they select.
```
